### PR TITLE
Add AWS functionality to workflow setup and forecast job

### PR DESCRIPTION
**Pull request by HenryRWinterbottom**
**Description**

This PR provides support for running UFS weather model forecasts, using the global-workflow, on the AWS Parallel Works (PW) platform. This PR accomplishes the following:

- Building the UFS weather model within the global-workflow build system;
- Running a UFS weather model atmosphere-only deterministic forecast using the global-workflow infrastructure.

Follow-up PRs will include the following:

- Adding documentation for building and using the AWS PW UFSWM spack-stack environment; this is heavily borrowed from Sadegh Sadeghi Tabas's presentation 03 May 2023;
- Addressing remaining `TODO` issues within this PR; several notes are made throughout as to how to better streamline the connections between the UFSWM and the global-workflow for the respective cloud platforms;
- Adding data-transfer capabilities within the global-workflow for cloud deployments;
- Testing additional UFSWM capabilities (e.g., coupling, etc.,) and resolutions (this PR has been tested at C48 and has not been configured for other cubed-sphere resolutions).

No dependencies other than the loading of the appropriate AWS PW modules are required. These are included in this PR. 

**Type of change**

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

**How Has This Been Tested?**

This PR has been tested as follows:
- The global-workflow fork/branch has been deployed to the AWS PW platform;
- The UFSWM is built using the global-workflow build system;
- A C48 atmosphere-only forecast has been run to completion. 
  
**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

### CODE REVIEW

File: jobs/rocoto/fcst.sh [hunk 1]
```diff
@@ -12,9 +12,10 @@ source "${HOMEgfs}/ush/preamble.sh"
 source "${HOMEgfs}/ush/detect_machine.sh"
 set +x
 source "${HOMEgfs}/ush/module-setup.sh"
-module use "${HOMEgfs}/sorc/ufs_model.fd/tests"
-module load modules.ufs_model.lua
-module load prod_util
```
Comment (WalterKolczynski-NOAA) [inline jobs/rocoto/fcst.sh:17]: You're still missing these module loads.

Comment (HenryRWinterbottom) [reply jobs/rocoto/fcst.sh:17]: Fixed.

File: sorc/build_ufs.sh [hunk 2]
```diff
@@ -32,10 +32,28 @@ COMPILE_NR=0
...
-cp "./modulefiles/ufs_common.lua" ./tests/ufs_common.lua
-cp "./modulefiles/ufs_common_spack.lua" ./tests/ufs_common_spack.lua
+if [[ "${MACHINE_ID}" != "noaacloud" ]]; then
+  ./tests/compile.sh "${MACHINE_ID}" "${MAKE_OPT}" "${COMPILE_NR}" "intel" "${CLEAN_BEFORE}" "${CLEAN_AFTER}"
+  mv "./tests/fv3_${COMPILE_NR}.exe" ./tests/ufs_model.x
+  mv "./tests/modules.fv3_${COMPILE_NR}.lua" ./tests/modules.ufs_model.lua
+  cp "./modulefiles/ufs_common.lua" ./tests/ufs_common.lua
+else
+  
+  if [[ "${PW_CSP}" == "aws" ]]; then
```
Comment (aerorahul) [inline sorc/build_ufs.sh]:
```suggestion
  if [[ "${PW_CSP:-}" == "aws" ]]; then
```

File: sorc/build_ufs.sh [hunk 3]
```diff
@@ -32,10 +32,28 @@ COMPILE_NR=0
 CLEAN_BEFORE=YES
 CLEAN_AFTER=NO
 
-./tests/compile.sh "${MACHINE_ID}" "${MAKE_OPT}" "${COMPILE_NR}" "intel" "${CLEAN_BEFORE}" "${CLEAN_AFTER}"
-mv "./tests/fv3_${COMPILE_NR}.exe" ./tests/ufs_model.x
-mv "./tests/modules.fv3_${COMPILE_NR}.lua" ./tests/modules.ufs_model.lua
-cp "./modulefiles/ufs_common.lua" ./tests/ufs_common.lua
-cp "./modulefiles/ufs_common_spack.lua" ./tests/ufs_common_spack.lua
```
Comment (WalterKolczynski-NOAA) [inline sorc/build_ufs.sh:39]: We might need this spack module soon. I wouldn't remove it.

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: sorc/build_ufs.sh (hunk 3 above)
Comment (HenryRWinterbottom) [reply sorc/build_ufs.sh:39]: @WalterKolczynski-NOAA I believe the UFS WM is carrying these around now. This was a hack in the original PR until @aerorahul pointed me to the UFS WM feature.

Comment (WalterKolczynski-NOAA) [reply sorc/build_ufs.sh:39]: This whole bit is happening inside the UFS. But just leave it, we can add the copy back when we need it.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**